<launch>
  <!-- Stomp Plugin for MoveIt -->
  <arg name="planning_plugin" value="stomp_moveit/StompPlannerManager" />

  <arg name="start_state_max_bounds_error" value="0.1" />
  <arg name="jiggle_fraction" value="0.05" />
  <!-- The request adapters (plugins) used when planning. ORDER MATTERS! -->
  <arg name="planning_adapters"
       default="default_planner_request_adapters/AddTimeParameterization
                default_planner_request_adapters/FixWorkspaceBounds
                default_planner_request_adapters/FixStartStateBounds
                default_planner_request_adapters/FixStartStateCollision
                default_planner_request_adapters/FixStartStatePathConstraints" />


  <param name="planning_plugin" value="$(arg planning_plugin)" />
  <param name="request_adapters" value="$(arg planning_adapters)" />
  <param name="start_state_max_bounds_error" value="$(arg start_state_max_bounds_error)" />
  <param name="jiggle_fraction" value="$(arg jiggle_fraction)" />

  <!-- Add MoveGroup capabilities specific to this pipeline -->
  <!-- <param name="capabilities" value="" /> -->

  <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml"/>
  <!-- Load parameters for all groups in movo_kg3.srdf -->
  <group ns="left_arm">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="left_arm" />
  </group>
  <group ns="right_arm">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="right_arm" />
  </group>
  <group ns="torso">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="torso" />
  </group>
  <group ns="left_gripper">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="left_gripper" />
  </group>
  <group ns="right_gripper">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="right_gripper" />
  </group>
  <group ns="head">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="head" />
  </group>
  <group ns="right_side">
    <rosparam command="load" file="$(find movo_7dof_moveit_config)/config/stomp_planning.yaml" />
    <param name="group_name" value="right_side" />
  </group>
</launch>
